src/main/java/io/camunda/testing/assertions/IncidentAssert.java

```diff
@@ -0,0 +1,214 @@
+package io.camunda.testing.assertions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.testing.filters.IncidentRecordStreamFilter;
+import io.camunda.testing.filters.StreamFilter;
+import io.camunda.zeebe.client.api.response.ActivatedJob;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.camunda.zeebe.protocol.record.value.ErrorType;
+import io.camunda.zeebe.protocol.record.value.IncidentRecordValue;
+import java.util.Optional;
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.StringAssert;
+import org.camunda.community.eze.RecordStreamSource;
+
+/** Assertions for incidents. An incident is identified by its incident key. */
+public class IncidentAssert extends AbstractAssert<IncidentAssert, Long> {
+
+  private final RecordStreamSource recordStreamSource;
+
+  public IncidentAssert(final long incidentKey, final RecordStreamSource recordStreamSource) {
+    super(incidentKey, IncidentAssert.class);
+    this.recordStreamSource = recordStreamSource;
+  }
+
+  /**
+   * Returns the incident key (useful for resolving the incident)
+   *
+   * @return key of the incident
+   */
+  public long getIncidentKey() {
+    return actual;
+  }
+
+  /**
+   * Asserts that the incident has the given error type
+   *
+   * @param expectedErrorType expected error type
+   * @return this {@link IncidentAssert}
+   */
+  public IncidentAssert hasErrorType(final ErrorType expectedErrorType) {
+    final IncidentRecordValue record = getIncidentCreatedRecordValue();
+    final ErrorType actualErrorType = record.getErrorType();
+
+    assertThat(actualErrorType)
+        .withFailMessage(
+            "Error type was not '%s' but was '%s' instead", expectedErrorType, actualErrorType)
+        .isEqualTo(expectedErrorType);
+
+    return this;
+  }
+
+  /**
+   * Asserts that the incident has the given error message
+   *
+   * @param expectedErrorMessage expected error message
+   * @return this {@link IncidentAssert}
+   */
+  public IncidentAssert hasErrorMessage(final String expectedErrorMessage) {
```

Comment:
❌  It would be cool if we could support string formatting in this assertion. Like AssertJ does in for example the `withFailMessage` below.

I put it as a blocker because I think we should have this before our first release. But I wouldn't mind if it is added at a later stage.

Reply:
Not sure what the benefit is. 

For me `hasErrorMessage("Unexpected Exception: %s" , exceptionMessage);` and `hasErrorMessage(String.format("Unexpected Exception: %s", exceptionMessage));` is kind of the same thing.

And of these two, I like the second better because then I know which syntax to use for placeholders in the string.